src/components/menu-surface/menu-surface.tsx

```diff
@@ -126,7 +126,9 @@ export class MenuSurface {
         }
 
         this.dismiss.emit();
-        this.preventClickEventPropagation();
+        if (!this.allowClicksElement) {
+            this.preventClickEventPropagation();
```

Comment:
This doesn't look like the right solution.

We already check (or try to check) if the click originated in the allowed element, and if it did, we return early:

<img width="636" alt="Screenshot of the code from line 107 until and including this code change" src="https://github.com/Lundalogik/lime-elements/assets/154725/31e11c35-b00f-4a9a-9e8a-e83f4da0d255">

Apparently, that code doesn't work correctly. Unless what you are really trying to do is something other than allow clicks on the supplied "allowed element"?

Reply:
Now that I look further into it, the original code works correctly except in the case of `limel-chipset`. Where I think this is falling down is, when a chip is clicked the `elementPath` does not include the `allowClicksElement`. The `allowClicksElement` is there and is correct, but the path shows simply as an array `[]` and this is causing the guard on `line 123` to be bypassed. 

At this point, `allowClicksElement` is `true`, but `clickedInAllowedElement` is `false`, the function does not return, and hence the issue is caused with the `preventClickEventPropogation` function being invoked. 

I guess perhaps I should look into the chipset component instead?

Reply:
> I guess perhaps I should look into the chipset component instead?

Yeah, if we either could fix the problem there, or at least find out enough to somehow make some kind of targeted workaround, that would be great!

Reply:
I'll give it a go

Reply:
That was a great call @adrianschmidt. The issue was still in the menu-surface, but higher up the chain and simply due to event.path not being the right way to get an element path from the shadowdom.

Reply:
Nice!

You forgot to remove the old attempt at a fix though 😊

Reply:
Ahahah, oops 😆